### PR TITLE
Fix bug #5604 - Unrecognised signal for touchable no longer on the screen 

### PR DESCRIPTION
**Pull request by eadz**
Fixes bug https://github.com/facebook/react-native/issues/5604 ("Redscreen error when TouchRelease triggered on a component no longer in the tree")

( maybe not ideally )

This issue is triggered by the following:

A Touch event on a component ( Keep Pressing )
A state/props update removes the component from the tree and render is performed
A Touch Release event. ( When you release the press on the component no longer there )

This fix adds an early return to the signal handling code if a RESPONDER_RELEASE signal happens when the responder has disappeared

### CODE REVIEW

File: Libraries/Components/Touchable/Touchable.js [hunk 1]
```diff
@@ -589,6 +589,9 @@ var TouchableMixin = {
    * @sideeffects
    */
   _receiveSignal: function(signal, e) {
+    if(!this.state.touchable.responderID && signal == Signals.RESPONDER_RELEASE) {
```
Comment (eadz) [inline Libraries/Components/Touchable/Touchable.js]: haha. Yes I am new to this.

**Comment by eadz** [comment]
cc @mkonicek

**Comment by rt2zz** [comment]
:+1:

**Comment by mkonicek** [comment]
Hey @eadz, thanks for the PR but I'd like to ask - have you seen the template when submitting this PR?
https://github.com/facebook/react-native/blob/master/PULL_REQUEST_TEMPLATE.md

Is it clear enough? Is there anything confusing / hard to understand about it?

Please add the necessary info so we can review this PR.

**Comment by eadz** [comment]
Hi @mkonicek 

It seems the Pull Request template was added after I submitted this PR, so no I didn't see it.

I am not skilled enough to work out how to automatically test this bug, but I could put together a sample component which demonstrates the issue if that helps?

Also, need to point out I'm a little out of my depth with understanding how the signal handling code works. I've found a bug, and I've found what I consider to be a possible fix, but I don't know if it's the correct fix nor how to test the fix in an automated way. Hoping for some help here.

**Comment by rt2zz** [comment]
These two errors should be `console.error`ed instead of thrown since they are not true exceptions and there is no way to try/catch as a consumer.

**Comment by mkonicek** [comment]
Ah, didn't realize this PR was from before when we added the template, sorry! Will take a look.

**Comment by mkonicek** [comment]
Since the error you see is:

    Unrecognized signal RESPONDER_RELEASE or state undefined for Touchable responder null

This looks like a reasonable fix.

@sahrens What do you think?

**Comment by sahrens** [comment]
Seems safe enough. I'm not exactly sure why the responderID is null rather than something being more broken about the state, if if this fixes the issue you're seeing it seems fine.

**Comment by sahrens** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**